res/skins/QMLDemo/HotcueButton.qml

```diff
@@ -63,16 +64,71 @@ Skin.Button {
     Popup {
         id: popup
 
+        dim: false
         modal: true
         focus: true
         closePolicy: Popup.CloseOnEscape | Popup.CloseOnPressOutsideParent
+        contentWidth: colorGrid.implicitWidth
+        contentHeight: colorGrid.implicitHeight + clearButton.implicitHeight
+
+        Grid {
+            id: colorGrid
+
+            columns: 4
+            spacing: 2
+
+            Repeater {
+                model: Mixxx.Config.getHotcueColorPalette()
+
+                Rectangle {
+                    height: 24
+                    width: 24
+                    color: modelData
+
+                    MouseArea {
+                        anchors.fill: parent
+                        onClicked: {
+                            hotcueColorControl.value = (parseInt(parent.color.r * 255) << 16) | (parseInt(parent.color.g * 255) << 8) | parseInt(parent.color.b * 255);
+                            popup.close();
+                        }
+                    }
+
+                }
+
+            }
+
+        }
+
+        Skin.ControlButton {
+            id: clearButton
+
+            anchors.top: colorGrid.bottom
+            anchors.left: parent.left
+            anchors.topMargin: 5
+            group: root.group
+            key: "hotcue_" + hotcueNumber + "_clear"
+            text: "Clear"
+            activeColor: Theme.deckActiveColor
+        }
+
+        enter: Transition {
+            NumberAnimation {
+                properties: "opacity"
+                from: 0
+                to: 1
+                duration: 100
+            }
+
+        }
+
+        exit: Transition {
+            NumberAnimation {
+                properties: "opacity"
+                from: 1
+                to: 0
+                duration: 100
+            }
```

Comment:
Prefer opacityanimator?

Reply:
Doesn't work for me, i get:

    Unable to assign HotcuePopup_QMLTYPE_33 to QQuickItem

Reply:
Thats strange. Does make a difference now anyways because the QQuickWidget is forcing the render loop to be single threaded anyways...